src/Audit.EntityFramework/DbContextHelper.cs

```diff
@@ -425,7 +425,9 @@ private static string GetTransactionId(DbTransaction transaction, string clientC
             var propIntTran = transaction.GetType().GetTypeInfo().GetProperty("InternalTransaction", BindingFlags.NonPublic | BindingFlags.Instance);
             object intTran = propIntTran?.GetValue(transaction);
             var propTranId = intTran?.GetType().GetTypeInfo().GetProperty("TransactionId", BindingFlags.NonPublic | BindingFlags.Instance);
-            var tranId = (int)(long)propTranId?.GetValue(intTran);
+            var tranId = 0;
+            if (propTranId != null)
+                tranId = (int)(long)propTranId.GetValue(intTran);
```

Comment:
It looks like we don't even need the casts

Reply:
That does work. Thank you.